plugin-flex-ts-template-v2/src/feature-library/dispositions/custom-components/DispositionTab/DispositionTab.tsx

```diff
@@ -50,23 +72,45 @@ const DispositionTab = (props: OwnProps) => {
     if (isNotesEnabled() && notes) {
       payload.notes = notes;
     }
-
+    payload.custom_attributes = { ...customAttributes };
     dispatch(updateDisposition(payload));
   };
 
   const updateStoreDebounced = debounce(updateStore, 250, { maxWait: 1000 });
 
   useEffect(() => {
```

Comment:
This `useEffect` has a lot of things going on. Here's a suggestion that might make this component a bit smaller, and easier to read. You could pull this into a custom react hook and just invoke it here. That way this whole `useEffect` block could be replaced with a single line.